### DISCUSSION TITLE
VSCode Extension

### DISCUSSION THREAD

**Discussion by SeriousM** [jotaen/xit#16] [Tooling]
A VSCode extension would be veeeery nice to have!

**Comment by petethepig** [comment]
Even just syntax highlighting would be really nice. I already adopted a similar system locally, but I have to use `.md` for syntax highlighting, which is not optimal.

**Comment by jotaen** [comment]
In case someone would be willing to create/contribute a VSCode extension, I’d be curious to hear how complicated it would be to account for [the idea of nested subitems/subtasks](https://github.com/jotaen/xit/discussions/2). This would require the highlighter to be recursive, which e.g. in the case of the Sublime Text engine turns out to be a bit tricky (due to limitations in their engine).

**Reply by tscpp** [reply]
It's almost impossible to get recursive syntax to work in textmate, however there is a thing called [semantic highlighting](https://code.visualstudio.com/api/language-extensions/semantic-highlight-guide) in vscode which provide semantic highlighting in addition to syntax highlighting.

**Comment by tscpp** [comment]
I completed my extension providing xit language support today. I loosened the strict requirements to provide smooth syntax highlighting while writing. See separate discussion (#20).

**Reply by vcsala** [reply]
I have used tscpp's extension (thanks, it is a great work!) and added some useful features. You can find the extension as xit! extended (still alpha release).

**Reply by jotaen** [reply]
Hey @vcsala, awesome, thanks for making this 🎉 . It’s [this one](https://marketplace.visualstudio.com/items?itemName=vcsala.xit-extended), right? I’m happy to add a link to the tool list on the [[x]it! website](https://xit.jotaen.net), to make it easier for other user to discover the extension. Let me know if or when you’d like that.

One tiny note about the logo: it would be cool if you could use a logo that looks different from [the “official” one](https://xit.jotaen.net/logo.png), to help users understand that your extension is an independent project. It’s totally okay if the logo looks somewhat similar, it just shouldn’t be the same.

**Reply by vcsala** [reply]
Hi Jotaen, thanks! It would be great if you could add it to the main page (and you are right, it is the vcsala/xit-extended extension).

Meantime, I am planning to release a bug fix version, I will change the logo too.
